app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt

```diff
@@ -0,0 +1,262 @@
+            val hasUsernameInFileName = filename.text.toString().isNotBlank()
+            // Use PasswordEntry to parse extras for username
+            val entry = PasswordEntry("PLACEHOLDER\n${extraContent.text}")
+            val hasUsernameInExtras = entry.hasUsername()
+            isEnabled = hasUsernameInFileName xor hasUsernameInExtras
+            isChecked = hasUsernameInExtras
+        }
+    }
+
+    /**
+     * Encrypts the password and the extra content
+     */
+    private fun encrypt(copy: Boolean = false) = with(binding) {
+        val editName = filename.text.toString().trim()
+        val editPass = password.text.toString()
+        val editExtra = extraContent.text.toString()
+
+        if (editName.isEmpty()) {
+            showSnackbar(resources.getString(R.string.file_toast_text))
+            return@with
+        }
+
+        if (editPass.isEmpty() && editExtra.isEmpty()) {
+            showSnackbar(resources.getString(R.string.empty_toast_text))
+            return@with
+        }
+
+        if (copy) {
+            copyPasswordToClipboard(editPass)
+        }
+
+        val data = Intent()
+        data.action = OpenPgpApi.ACTION_ENCRYPT
+
+        // EXTRA_KEY_IDS requires long[]
+        val longKeys = keyIDs.map { it.toLong() }
+        data.putExtra(OpenPgpApi.EXTRA_KEY_IDS, longKeys.toLongArray())
+        data.putExtra(OpenPgpApi.EXTRA_REQUEST_ASCII_ARMOR, true)
+
+        // TODO Check if we could use PasswordEntry to generate the file
+        val content = "$editPass\n$editExtra"
+        val inputStream = ByteArrayInputStream(content.toByteArray())
+        val outputStream = ByteArrayOutputStream()
+
+        val path = when {
+            // If we allowed the user to edit the relative path, we have to consider it here instead
+            // of fullPath.
+            category.isEnabled -> {
+                val editRelativePath = category.text.toString().trim()
+                if (editRelativePath.isEmpty()) {
+                    showSnackbar(resources.getString(R.string.path_toast_text))
+                    return
+                }
+                val passwordDirectory = File("$repoPath/${editRelativePath.trim('/')}")
+                if (!passwordDirectory.exists() && !passwordDirectory.mkdir()) {
+                    showSnackbar("Failed to create directory ${editRelativePath.trim('/')}")
+                    return
+                }
+
+                "${passwordDirectory.path}/$editName.gpg"
+            }
+            else -> "$fullPath/$editName.gpg"
+        }
+
+        lifecycleScope.launch(Dispatchers.IO) {
+            api?.executeApiAsync(data, inputStream, outputStream) { result ->
+                when (result?.getIntExtra(OpenPgpApi.RESULT_CODE, OpenPgpApi.RESULT_CODE_ERROR)) {
+                    OpenPgpApi.RESULT_CODE_SUCCESS -> {
+                        try {
+                            // TODO This might fail, we should check that the write is successful
```

Comment:
This might be a good time to resolve this todo. How about a dialog and an explicit `setResult(RESULT_CANCELED)` in the `catch`?